src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -67,10 +61,16 @@
 
     private static final String SYSTEM_DEFAULT_CHARSET = "SYSTEM_DEFAULT";
 
-    /** Number of times we will show launch diagnostics in a newly encountered workspace before going mute to save resources. */
+    /**
+     * Whether to stream stdio from the wrapper script, which should normally not print any.
+     * Copying output from the controller process consumes a Java thread, so we want to avoid it generally.
+     * If requested, we can do this to assist in diagnosis.
+     * (For example, if we are unable to write to a workspace due to permissions,
+     * we would want to see that error message.)
+     */
     @SuppressWarnings("FieldMayBeFinal")
     // TODO use SystemProperties if and when unrestricted
-    private static int NOVEL_WORKSPACE_DIAGNOSTICS_COUNT = Integer.getInteger(BourneShellScript.class.getName() + ".NOVEL_WORKSPACE_DIAGNOSTICS_COUNT", 10);
+    private static boolean LAUNCH_DIAGNOSTICS = Boolean.getBoolean(BourneShellScript.class.getName() + ".LAUNCH_DIAGNOSTICS");
```

Comment:
Any idea of whether having this show up the first 10 times was useful in practice? I can't think of any cases where I saw output from the wrapper in the past, but maybe I just didn't recognize it when it happened.

Reply:
There are some cases. For example, if you messed up

```groovy
withEnv("PATH=${tool 'whatever'}/bin") { // meant to use PATH+WHATEVER or explicitly append/prepend
  sh 'oops'
}
```

then the wrapper script (pending #92) would not start, yet you would not get an explanation of why in the build log. Another case is that you messed something up with `withDockerContainer` so that the workspace volume lacked write access by the effective user ID, so the attempted redirect to the output file would fail. But in the normal case that the wrapper runs (even if you have fatal errors in the user process), piping stdio from the wrapper process merely wasted a thread.

The `encounteredPaths` / `novel` trick being deleted here was an attempt to compromise: when running the launcher the first few times on a particular workspace that might be set up poorly, print diagnostics, but subsequently skip it for efficiency. Unfortunately that trick does not work well in a `Cloud` world—you can expect _every_ workspace to be novel.

Anyway now with the backgrounding it is really impossible to track wrapper process output, so I have kept the diagnostics available only as a global option. With the diagnostic flag off by default, this means that a misconfigured path/workspace/etc. will result in the -2 exit code and the message

> process apparently never started in …

but no further details. Ideally there would be an easy way to enable launch diagnostics (and now, disable backgrounding) on the fly, or at least an option to `BourneShellScript` + `ShellStep` to enable it from the `Jenkinsfile`.

Reply:
Ok, thanks for elaborating. The current behavior seems fine to me. Maybe we could have the "process apparently never started" message also mention something like "consider temporarily enabling diagnostic info through the script console by setting ...".